TextAnalysis/src/langanal/word/processing/WordProcessing.java

```diff
@@ -0,0 +1,239 @@
+			for(String curWord : defWordList1){
+				definitionWords1.add(curWord);
+			}
+		}
+
+		//Fills definitionWords2
+		for(String fullDefW2 : word2.getDefinitions()){
+			ArrayList<String> defWordList2 = definitionToWords(fullDefW2);
+
+			for(String curWord : defWordList2){
+				definitionWords2.add(curWord);
+			}
+		}
+
+		//Clears common words
+		for(int i = 0;i<definitionWords1.size();i++){
+
+		}
+		//Checks for repetitions
+		for(String persistWord : definitionWords1){
+			for(String checkingWord : definitionWords2){
+				if(persistWord.equals(checkingWord)){
+					if(!isCommonWord(persistWord)){
+						definitionRepeats++;
+					}
+				}
+			}
+		}
+
+		return definitionRepeats;
+	}
+
+	/**
+	 * Returns an ArrayList of words that are formatted for checkDefinition()
+	 * @param definition
+	 * @return
+	 */
+	private static ArrayList<String> definitionToWords(String definition){
+		ArrayList<String> returnList = new ArrayList<String>();
+
+		for(String word : definition.replaceAll("[^a-zA-Z ]", "").toLowerCase().trim().split(" ")){
+			returnList.add(word.trim());
+		}
+		return returnList;
+	}
+
+	/**
+	 * Determines whether the inputted word is a common one
+	 * 
+	 * @param word
+	 * @return Boolean whether the inputted word is a common word
+	 */
+	private static Boolean isCommonWord(String word){
+		boolean isIn = false;
+
+		for(String s : commonWords){
+			if(word.equals(s)){
+				isIn = true;
+			}
+		}
+
+		return isIn;
+	}
+
+	/**
+	 * Checks to see if the POS is the same between two words
+	 * 
+	 * @param word1
+	 * @param word2
+	 * @return 1 if they share the same POS, 0 if they do not share the same POS
+	 */
+	public static int checkPOS(Word word1, Word word2){
+		if(word1.getPOS().equals(word2.getPOS())){
+			return 1;
+		}else{
+			return 0;
+		}
+	}
+
+	/**
+	 * Checks to see if two words share any synonyms
+	 * @param word1
+	 * @param word2
+	 * @return The # of shared synonyms
+	 */
+	public static int checkSynonyms(Word word1, Word word2){
+		ArrayList<String> synonymsWord1 = new ArrayList<String>(word1.getSynonyms().size());
+		ArrayList<String> synonymsWord2 = new ArrayList<String>(word2.getSynonyms().size());
+
+		int repeatedSynonyms = 0; //holds the # of repeated pairs of similarity words
+
+		for(String s : cleanSynonyms(word1.getSynonyms())){
+			synonymsWord1.add(s);
+		}
+		for(String s : cleanSynonyms(word2.getSynonyms())){
+			synonymsWord2.add(s);
+		}
+
+		for(String curSynonymWord1 : synonymsWord1){
+			for(String curSynonymWord2 : synonymsWord2) {
+				if(curSynonymWord1.equals(curSynonymWord2)){
+					repeatedSynonyms++;
+				}
+			}
+		}
+		return repeatedSynonyms;
+	}
+
+	/**
+	 * Cleans and formats the ArrayList for the checkSynonyms function
+	 * @param list
+	 * @return cleaned and formatted list
+	 */
+	private static ArrayList<String> cleanSynonyms(LinkedList<String> list){
+		ArrayList<String> returnList = new ArrayList<String>();
+
+		for(String s : list){
+			for(String x : s.split(" ")){
```

Comment:
Minor, but how does this perform on compound words that lack a hyphen? Does it split the words?